server/routes/products.js

```diff
@@ -0,0 +1,93 @@
+const db = require("../models");
+const express = require("express");
+const router = express.Router();
+const authMiddleware = require("../middlewares/authMiddleware");
+const {
+  validate,
+  listIdCheck,
+  productIdCheck,
+} = require("../middlewares/validate");
+const ProductModel = require("../models/productModel");
+const ListProductModel = require("../models/listProductModel");
+
+const giveServerError = (res) =>
+  res.status(500).send({ errors: [{ msg: "Server error" }] });
+
+const getProduct = async (req, res) => {
+  try {
+    const result = await ProductModel.findOne({
+      where: { id: parseInt(req.params.productId) },
+      attributes: [
+        "id",
+        "current_price",
+        "previous_price",
+        "name",
+        "image_url",
+        "link",
+        "company",
+        "is_still_available",
+      ],
+    });
+    if (result == null) {
+      res.status(400).send({ errors: [{ msg: "No product with given ID." }] });
+      return;
+    }
+    res.status(200).send(result);
+  } catch (error) {
+    console.error(error);
+    giveServerError(res);
+  }
+};
+
+//Removes product, then checks to see if the product still exists in other lists. If it doesn't, the product is removed
+//from the database.
+const deleteProduct = async (req, res) => {
+  let transaction = null;
+  const productId = parseInt(req.params.productId);
+  const listId = parseInt(req.params.listId);
+  try {
+    transaction = await db.transaction();
+    await db.query(
+      `
+    DELETE FROM "ListProducts"
+    USING "UserLists"
+    WHERE (
+        "ListProducts".list_id = "UserLists".id
+        AND "ListProducts".product_id = ${productId} AND "ListProducts".list_id = ${listId}
+        AND "UserLists".user_id = ${req.user.id}
+    )
```

Comment:
this can be updated to sequelize query

Reply:
Unfortunately the `Model.destroy()` method does not allow for the use of associations and we need to make sure that the list we are deleting belongs to the user. Otherwise, the user can delete anyone's list.